src/analytics.js

```diff
@@ -244,6 +245,155 @@ const Analytics = {
 
 		return await Analytics.client;
 	},
+	audiences: [],
+	updateAudiences: async () => {
+		// Take a clone of the current audiences to check if they changed later.
+		const oldAudienceIds = Analytics.audiences.slice().sort();
+
+		// Get attributes.
+		const attributes = await prepareAttributes( getAttributes() );
+		// Get metrics.
+		const metrics = await prepareMetrics( getMetrics() );
+		// Get endpoint.
+		const endpoint = Analytics.getEndpoint();
+
+		// Aggregate data into a queryable object.
+		const fieldData = {
+			attributes,
+			metrics,
+			endpoint,
+		};
+
+		// Store derived field values for quicker lookups.
+		const fieldValues = {};
+
+		// Audience ID store.
+		const audienceIds = [];
+
+		// Map audience configurations to IDs.
+		for ( const aid in Audiences ) {
+			const { id, config } = Audiences[ aid ];
```

Comment:
Could we pull this whole inner blocker into it's own function? Seems the job of "match the user endpoints (etc) to our Audience objects should be able to be self-contained in a function?

Reply:
Ah looking below, perhaps this function does both: fetch the user's audiences and also trigger an "audiences updated" event if they have changed?

Reply:
Yeah I was thinking of maybe splitting it up but I'm not sure how much more readable it would be beyond less indentation. It was just easier for me to write it this way first.